### ISSUE TITLE
Prevent overwriting events when persisting

### ISSUE THREAD

**Issue by Fours** [akka/akka-persistence-dynamodb-1.x#64]
Could use a condition expression to stop an overwrite and instead return a rejection: http://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Expressions.ConditionExpressions.html

This would be helpful in the exceptional scenario of two persistent actors running with the same persistence id, trying to save an event with the same seqNum - I would rather one fail to persist than overwrite an existing item.

Thoughts?

**Comment by mallize** [comment]
Agreed. I would never want to overwrite an event. Should this end up with a call to onPersistFailure which would allow the developer to decide how to handle this?

**Comment by calvinlfer** [comment]
+1
This would help detect split brain scenarios and prevent corruption of your data where you use Akka Cluster Sharding and the constraint of having only a single Persistent Actor with a given ID X within a cluster is violated
